catalog/spatial/registry/registry-rest-endpoint/src/main/java/org/codice/ddf/registry/rest/endpoint/RegistryRestEndpoint.java

```diff
@@ -0,0 +1,171 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+
+package org.codice.ddf.registry.rest.endpoint;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.apache.commons.lang.StringUtils;
+import org.codice.ddf.registry.federationadmin.service.FederationAdminException;
+import org.codice.ddf.registry.federationadmin.service.FederationAdminService;
+import org.codice.ddf.registry.federationadmin.service.RegistryPublicationService;
+import org.codice.ddf.registry.rest.endpoint.report.RegistryReportHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.jknack.handlebars.Handlebars;
+import com.github.jknack.handlebars.Template;
+import com.github.jknack.handlebars.io.ClassPathTemplateLoader;
+
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.RegistryPackageType;
+
+@Path("/")
+public class RegistryRestEndpoint {
+    private static final Logger LOGGER = LoggerFactory.getLogger(RegistryRestEndpoint.class);
+
+    private FederationAdminService federationAdminService;
+
+    private ClassPathTemplateLoader templateLoader;
+
+    private RegistryPublicationService registryPublicationService;
+
+    private RegistryReportHelper reportHelper;
+
+    public RegistryRestEndpoint() {
+        templateLoader = new ClassPathTemplateLoader();
+        templateLoader.setPrefix("/templates");
+        templateLoader.setSuffix(".hbt");
+    }
+
+    @Path("/{registryId}/publication/{sourceId}")
+    @POST
+    public Response publish(@PathParam("registryId") String registryId,
+            @PathParam("sourceId") String destinationId) {
+        if (StringUtils.isBlank(registryId) || StringUtils.isBlank(destinationId)) {
+            return Response.status(Response.Status.BAD_REQUEST)
+                    .entity("Registry and destination ID cannot be blank")
```

Comment:
They're both saying the same thing. Neither the registry id nor the destination id can be blank. The if checks if either is blank and the message says both cannot be blank. The check has been changed as suggested below.